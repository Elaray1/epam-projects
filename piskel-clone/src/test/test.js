const functionsArr = require('./functionsToTest');

const convertRgbToHex = functionsArr.convertRgbToHex;

describe('Color Change Check (from RGB to HEX)', () => {
  test('transform RGB format in HEX format (white)', () => {
    expect(convertRgbToHex(255, 255, 255)).toBe('#ffffff');
    expect(convertRgbToHex(255, 255, 255)).toMatchSnapshot();
  });

  test('transform RGB format in HEX format (black)', () => {
    expect(convertRgbToHex(0, 0, 0)).toBe('#000000');
    expect(convertRgbToHex(0, 0, 0)).toMatchSnapshot();
  });

  test('transform RGB format in HEX format (brown)', () => {
    expect(convertRgbToHex(127, 64, 15)).toBe('#7f400f');
    expect(convertRgbToHex(127, 64, 15)).toMatchSnapshot();
  })

  test('transform RGB format in HEX format (dark grey)', () => {
    expect(convertRgbToHex(47, 47, 47)).toBe('#2f2f2f');
    expect(convertRgbToHex(47, 47, 47)).toMatchSnapshot();
  })

  test('transform RGB format in HEX format (grey)', () => {
    expect(convertRgbToHex(77, 77, 77)).toBe('#4d4d4d');
    expect(convertRgbToHex(77, 77, 77)).toMatchSnapshot();
  })
});
