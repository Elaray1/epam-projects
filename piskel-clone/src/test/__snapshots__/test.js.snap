// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`convertRgbToHex transform RGB format in HEX format (black) 1`] = `"#7f400f"`;

exports[`convertRgbToHex transform RGB format in HEX format (black) 2`] = `"#2f2f2f"`;

exports[`convertRgbToHex transform RGB format in HEX format (black) 3`] = `"#4d4d4d"`;

exports[`convertRgbToHex transform RGB format in HEX format 1`] = `"#ffffff"`;

exports[`convertRgbToHex transform RGB format in HEX format 2`] = `"#000000"`;
